src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/LifetimeValidationError.cs

```diff
@@ -50,33 +34,45 @@ internal override Exception GetException()
         {
             if (ExceptionType == typeof(SecurityTokenNoExpirationException))
             {
-                return new SecurityTokenNoExpirationException(MessageDetail.Message);
+                var exception = new SecurityTokenNoExpirationException(MessageDetail.Message, InnerException);
+                exception.SetValidationError(this);
+                return exception;
             }
             else if (ExceptionType == typeof(SecurityTokenInvalidLifetimeException))
             {
-                return new SecurityTokenInvalidLifetimeException(MessageDetail.Message)
+                var exception = new SecurityTokenInvalidLifetimeException(MessageDetail.Message, InnerException)
                 {
-                    NotBefore = _notBefore,
-                    Expires = _expires
+                    NotBefore = NotBefore,
+                    Expires = Expires
                 };
+                exception.SetValidationError(this);
+                return exception;
             }
             else if (ExceptionType == typeof(SecurityTokenNotYetValidException))
             {
-                return new SecurityTokenNotYetValidException(MessageDetail.Message)
+                var exception = new SecurityTokenNotYetValidException(MessageDetail.Message, InnerException)
                 {
-                    NotBefore = _notBefore
+                    NotBefore = NotBefore
                 };
+                exception.SetValidationError(this);
+                return exception;
             }
             else if (ExceptionType == typeof(SecurityTokenExpiredException))
             {
-                return new SecurityTokenExpiredException(MessageDetail.Message)
+                var exception = new SecurityTokenExpiredException(MessageDetail.Message, InnerException)
                 {
-                    Expires = _expires
+                    Expires = Expires
                 };
+                exception.SetValidationError(this);
+                return exception;
             }
             else
                 return base.GetException(ExceptionType, null);
         }
+
+        protected DateTime NotBefore { get; set; }
```

Comment:
Why protected?
No need for 'set' as ctor sets value.

Reply:
We keep it protected so that subclasses can set this value.

Reply:
Removed.